src/hotspot/share/cds/filemap.cpp

```diff
@@ -926,15 +926,59 @@ void FileMapInfo::log_paths(const char* msg, int start_idx, int end_idx) {
   }
 }
 
+bool FileMapInfo::is_jar_suffix(const char* filename) {
+  const char* dot = strrchr(filename, '.');
+  if (strcmp(dot + 1, "jar") == 0) {
```

Comment:
What if there is no dot? We need a null check.

Reply:
Added a null check.